echopop/utils/batch_load.py

```diff
@@ -0,0 +1,608 @@
+        - `latitude` (float): Latitude (degrees).
+        - `longitude` (float): Longitude (degrees).
+        - `transect_spacing` (float): Distance between transects (nmi).
+        - `NASC_no_age` (float): Age-2+ NASC.
+        - `NASC_all_ages` (float): Age-1+ NASC.
+        - `haul_num` (int): The trawl/haul number associated with each transect interval.
+        - `stratum_num` (Optional[int]): Length-based (KS clustered) stratum numbers that are added
+        when `include_stratum = True` and `strata = 'ks'`.
+        - `stratum_inpfc` (Optional[int]): Latitude-based (INPFC) stratum numbers that are added
+        when `include_stratum = True` and `strata = 'inpfc'`.
+
+    Notes
+    ----------
+    The string pattern for `transect_pattern` requires a consistent filename format that can be
+    readily parsed by `read_echoview_exports`. The default value for `transect_pattern`
+    (``r'T(\\d+)'``) enables the function to parse all numbers that trail the letter "T" in the base
+    filename. For example, an example file of "C:/Path/User/Data/random_12_V34-T56-A78_G9.csv" would
+    yield a transect number of '56' since it trails the "T" and does not accidentally use other
+    numbers in the string. However, a filename like
+    "C:/Path/User/Data/randomT_12_T34-T56-T78_G9.csv" would detect multiple transect numbers ('34',
+    '56', and '78') since numbers trail the letter "T" in three places. Therefore, it is important
+    to ensure that embedded transect numbers are differentiable from other digits that may appear
+    in the base filename.
+    """
+
+    # Get NASC export settings
+    export_settings = configuration_dict["nasc_exports"]
+
+    # Check that save directory exists
+    # ---- Get save file directory
+    if save_directory is None:
+        save_file_directory = export_settings["nasc_export_directory"]
+    elif not isinstance(save_directory, (str, Path)):
+        raise TypeError(f"Save directory for NASC file {save_directory} must be a `Path` or `str`.")
+    else:
+        save_file_directory = save_directory
+    # ---- Drop prepended "/" or "\\" to avoid using an absolute path
+    if save_file_directory.startswith("/") or save_file_directory.startswith("\\"):
+        save_file_directory = save_file_directory[1:]
+    # ---- Create filepath
+    save_folder = str(Path(configuration_dict["data_root_dir"]) / save_file_directory)
+    # ---- Validate existence
+    if not Path(save_folder).exists():
+        raise FileNotFoundError(f"Save directory for NASC file ({save_folder}) does not exist.")
+
+    # Check that directory exists
+    # ---- Get export file directory
+    export_file_directory = export_settings["export_file_directory"]
+    # ---- Drop prepended "/" or "\\" to avoid using an absolute path
+    if export_file_directory.startswith("/") or export_file_directory.startswith("\\"):
+        export_file_directory = export_file_directory[1:]
```

Comment:
I think it may be a good idea to not leave this "back door" open for users to be able to separately put in `file_directory` and `save_directory`, and force all operations to be via the config files. This way we can always (at least within reasonable ranges) trace back on which config file ingest what data and export what files. So essentially I think **less** flexibility here is better, so that we prevent accidental shorthands or things people forget to change.